.golangci.yml

```diff
@@ -0,0 +1,139 @@
+linters-settings:
+  dupl:
+    threshold: 100
+  funlen:
+    lines: 100
+    statements: 50
+  gci:
+    local-prefixes: fybrik.io/data-movement-controller
+  goconst:
+    min-len: 6
+    min-occurrences: 2
+  gocritic:
+    enabled-tags:
+      - diagnostic
+      - experimental
+      - opinionated
+      - performance
+      - style
+    disabled-checks:
+      - dupImport # https://github.com/go-critic/go-critic/issues/845
+      - ifElseChain
+      - octalLiteral
+      - whyNoLint
+      - wrapperFunc
+  gocyclo:
+    min-complexity: 15
+  goimports:
+    local-prefixes: fybrik.io/data-movement-controller
+  gomnd:
+    settings:
+      mnd:
+        # don't include the "operation" and "assign"
+        checks: argument,case,condition,return
+        ignored-numbers: 0,1,2,3,404
+        ignored-functions: strings.SplitN
+
+  govet:
+    check-shadowing: true
+  lll:
+    line-length: 140
+  misspell:
+    locale: US
+  nolintlint:
+    allow-leading-space: true # don't require machine-readable nolint directives (i.e. with no leading space)
+    allow-unused: false # report any unused nolint directives
+    require-explanation: false # don't require an explanation for nolint directives
+    require-specific: false # don't require nolint directives to be specific about which linter is being skipped
+  gosec:
+    # Exclude generated files
+    exclude-generated: true
+  revive:
+    # see https://github.com/mgechev/revive#available-rules for details.
+    ignore-generated-header: true
+    severity: warning
+    rules:
+      - name: indent-error-flow
+        severity: warning
+      - name: add-constant
+        severity: warning
+        arguments:
+          - maxLitCount: "3"
+            allowStrs: '"","-"'
+            allowInts: "0,1,2,404"
+            allowFloats: "0.0,0.,1.0,1.,2.0,2."
+
+linters:
+  disable-all: true
+  enable:
+    - bodyclose
+    - deadcode
+    - dogsled
+    - dupl
+    - errcheck
+    - exportloopref
+    - funlen
+#    - gochecknoinits
+    - goconst
+    - gocritic
+    - gocyclo
+    - gofmt
+    - goimports
+    - gomnd
+    - goprintffuncname
+    - gosec
+    - gosimple
+    - govet
+    - ineffassign
+    - lll
+    - nakedret
+    - noctx
+    - nolintlint
+    - staticcheck
+    - structcheck
+    - stylecheck
+    - typecheck
+    - unconvert
+    - unparam
+    - unused
+    - varcheck
+    - whitespace
+    - revive
+
+  # don't enable:
+  # - asciicheck
+  # - depguard
+  # - scopelint
+  # - gochecknoglobals
+  # - gocognit
+  # - godot
+  # - godox
+  # - misspell
+  # - goerr113
+  # - interfacer
+  # - maligned
+  # - nestif
+  # - prealloc
+  # - testpackage
+  # - wsl
+issues:
+  exclude-rules:
+    - path: _test\.go
+      linters:
+       - dupl
+       - revive
+    - path: pkg/test/
+      linters:
+       - revive
+       - gomnd
+       - gocritic
+    - linters:
```

Comment:
A path is not necessary because I want this linter to work on the whole repo and not on a single file, it's suppose to ignore length of long comments.
The reason I added it is to ignore the long kubebuilder markers.

Reply:
>  I want this linter to work on the whole repo
I think you ignored the linter, you added it under `exclude-rule`

Reply:
That what I meant, I'll rephrase it:
I want the exception rule of this linter to work on the whole repo, meaning on all the files long comment will not raise a linter warning.

Reply:
therefore, maybe will be better to disable this linter for now